cmd/plugin/cmd/delete.go

```diff
@@ -138,9 +159,190 @@ func runDelete() error {
+	}
+
+	log.Info("Waiting for CAPI Operator manifests to be removed...")
+
+	if err := wait.ExponentialBackoff(opts, func() (bool, error) {
+		ready := true
+		for i := range d.providers {
+			if done, err := deleteProviders(ctx, cl, d.providers[i], ctrlclient.MatchingFieldsSelector{
+				Selector: fields.SelectorFromSet(d.selectors[i]),
+			}); err != nil {
+				return false, err
+			} else {
+				ready = ready && done
+			}
+		}
+
+		return ready, nil
+	}); err != nil {
+		return fmt.Errorf("cannot remove provider: %w", err)
+	}
+
+	return nil
+}
+
+func selectorFromProvider(provider string) (fields.Set, error) {
+	var name, namespace string
+
+	parts := strings.Split(provider, ":")
+	switch len(parts) {
+	case 0 | 3:
+	case 1:
+		name = parts[0]
+	case 2:
+		name, namespace = parts[0], parts[1]
+	default:
+		return nil, fmt.Errorf("invalid provider format: %s", provider)
+	}
+
+	selector := fields.Set{}
+
+	if name != "" {
+		selector["metadata.name"] = name
+	}
+
+	if namespace != "" {
+		selector["metadata.namespace"] = namespace
+	}
+
+	return selector, nil
+}
+
+func deleteProviders(ctx context.Context, client ctrlclient.Client, providerList genericProviderList, selector ctrlclient.MatchingFieldsSelector) (bool, error) {
+	//nolint:forcetypeassert
+	providerList = providerList.DeepCopyObject().(genericProviderList)
+	ready := true
+
+	gvks, _, err := scheme.ObjectKinds(providerList)
+	if err != nil {
+		log.Error(err, "Kind is not registered in provider list")
+		return false, err
+	}
+
+	gvk := gvks[0]
+
+	if err := client.List(ctx, providerList, selector); meta.IsNoMatchError(err) || apierrors.IsNotFound(err) {
+		return true, nil
+	} else if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to list providers to delete, %#v", err))
+		return false, err
+	}
+
+	for _, provider := range providerList.GetItems() {
+		log.Info(fmt.Sprintf("Deleting %s %s/%s", provider.GetType(), provider.GetName(), provider.GetNamespace()))
+
+		provider, ok := provider.(genericProvider)
+		if !ok {
+			log.Info(fmt.Sprintf("Expected to get GenericProvider for %s", gvk))
+			continue
+		}
+
+		if err := client.DeleteAllOf(ctx, provider, ctrlclient.InNamespace(provider.GetNamespace())); err != nil {
+			return false, fmt.Errorf("unable to issue delete for %s: %w", gvk, err)
+		}
+
+		if deleteOpts.includeNamespace {
+			if strings.HasPrefix(provider.GetNamespace(), "kube-") || provider.GetNamespace() == "default" {
+				log.Info(fmt.Sprintf("Skipping system namespace %s", provider.GetNamespace()))
+				continue
+			}
+
+			ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: provider.GetNamespace()}}
+			if err := client.Delete(ctx, ns); ctrlclient.IgnoreNotFound(err) != nil {
+				return false, fmt.Errorf("unable to issue delete for Namespace %s: %w", provider.GetNamespace(), err)
+			}
+		}
+	}
+
+	if len(providerList.GetItems()) > 0 {
+		log.Info(fmt.Sprintf("%d items remaning...", len(providerList.GetItems())))
+		return false, nil
+	}
```

Comment:
I suggest to add 
else {
    log.Info("All required providers were deleted")
}

Reply:
Added below.